### ISSUE TITLE
[terra-clinical-data-grid] Add accessibility for empty cells 

### ISSUE THREAD

**Issue by sbulley09** [cerner/terra-clinical#703]
# Feature Request

## Description

Selectable data grid cells which do not have content currently get flagged for [accessibility violations](https://dequeuniversity.com/rules/axe/3.5/button-name?application=axeAPI).  Since empty cells are common, we would like a method by which we can be compliant with the accessibility standard. Perhaps an aria-label option in the cells properties.

## Additional Context / Screenshots

This screenshot displays a scenario where there is no text in a cell, but they violate the accessibility rules.
<img width="1081" alt="accessibility1" src="https://user-images.githubusercontent.com/39015681/94204699-5a76da00-fe87-11ea-9aca-c63e902e41c6.png">

## @ Mentions

I've spoken briefly with @tbiethman about this enhancement.
